packages/smarthr-ui/src/hooks/useDecorators.ts

```diff
@@ -0,0 +1,27 @@
+import { ReactNode, useMemo } from 'react'
+
+export type DecoratorsType<T extends string> = {
+  [K in T]?: DecoratorType
+}
+
+export type DecoratorType = (text: string) => ReactNode
+
+export const useDecorators = <T extends string>(
+  defaultTexts: { [K in T]: string },
+  decorators: DecoratorsType<T> | undefined,
+) =>
+  useMemo(() => {
+    if (!decorators) {
+      return defaultTexts
+    }
+
+    const decorated = {} as { [K in T]: ReactNode }
```

Comment:
asちょっと怖い気がするので
```suggestion
    const decorated: { [K in T]: ReactNode } = {}
```
とか、satisfiesとかで代用できないんでしたっけ

Reply:
変数に型として設定してしまうと、 `{}` は明らかに型違反なので駄目っすね...

Reply:
例えば

```
const decorated: { [key: string]: ReactNode } = {}

....

return decorated as { [K in T]: ReactNode }
```

とすればもうちょいわかりやすいかもですがあまり意味はなさそう...